drivers/touch/iqs5xx/iqs5xx.c

```diff
@@ -0,0 +1,343 @@
+		ret = i2c_write_dt(client, mbuf, mlen);
+		if (ret >= 0)
+			return 0;
+
+		k_sleep(K_USEC(200));
+	}
+
+	if (ret >= 0)
+		ret = -EIO;
+
+	LOG_ERR("Failed to write to address 0x%04X: %d\n", reg, ret);
+
+	return ret;
+}
+
+static int iqs5xx_write_word(const struct i2c_dt_spec *client, uint16_t reg, uint16_t val)
+{
+	uint16_t val_buf = sys_cpu_to_be16(val);
+
+	return iqs5xx_write_burst(client, reg, &val_buf, sizeof(val_buf));
+}
+
+static int iqs5xx_write_byte(const struct i2c_dt_spec *client, uint16_t reg, uint8_t val)
+{
+	return iqs5xx_write_burst(client, reg, &val, sizeof(val));
+}
+
+static void setup_int(const struct device *dev, bool enable)
+{
+	const struct iqs5xx_dev_config *cfg = dev->config;
+	gpio_flags_t flags = enable ? GPIO_INT_EDGE_RISING : GPIO_INT_DISABLE;
+
+	gpio_pin_interrupt_configure_dt(&cfg->rdy_gpio, flags);
+}
+
+static void iqs5xx_gpio_callback(const struct device *dev, struct gpio_callback *cb, uint32_t pins)
+{
+	struct iqs5xx_data *data = CONTAINER_OF(cb, struct iqs5xx_data, gpio_cb);
+
+	setup_int(data->dev, false);
+	k_work_submit(&data->work);
+}
+
+static void iqs5xx_work_cb(struct k_work *work)
+{
+	struct iqs5xx_data *data = CONTAINER_OF(work, struct iqs5xx_data, work);
+
+	const struct iqs5xx_dev_config *cfg = data->dev->config;
+
+	iqs5xx_read_burst(&cfg->i2c, IQS5XX_GEST_EV0, &data->regmap, sizeof(data->regmap));
+
+	iqs5xx_write_byte(&cfg->i2c, IQS5XX_END_COMM, 0);
+
+	/* Small delay required as the iqs twiddles the rdy pin. */
+	k_sleep(K_USEC(50));
+
+#ifdef CONFIG_TOUCH_DEBUG
+	if (data->regmap.gesture_event[0]) {
+		switch (data->regmap.gesture_event[0] & 0b00111111) {
+		case IQS5XX_SINGLE_TAP:
+			LOG_INF("Single Tap");
+			break;
+		case IQS5XX_TAP_AND_HOLD:
+			LOG_INF("Tap And Hold");
+			break;
+		case IQS5XX_SWIPE_X_NEG:
+			LOG_INF("Swipe X negative");
+			break;
+		case IQS5XX_SWIPE_X_POS:
+			LOG_INF("Swipe X positive");
+			break;
+		case IQS5XX_SWIPE_Y_NEG:
+			LOG_INF("Swipe Y negative");
+			break;
+		case IQS5XX_SWIPE_Y_POS:
+			LOG_INF("Swipe Y positive");
+			break;
+		default:
+			LOG_INF("%d", data->regmap.gesture_event[0]);
+		}
+	}
+
+	if (data->regmap.gesture_event[1]) {
+		switch (data->regmap.gesture_event[1] & 0b00000111) {
+		case IQS5XX_TWO_FINGER_TAP:
+			LOG_INF("Two Finger Tap");
+			break;
+		case IQS5XX_SCROLL:
+			LOG_INF("Scroll");
+			break;
+		case IQS5XX_ZOOM:
+			LOG_INF("Zoom");
+			break;
+		default:
+			LOG_INF("%d", data->regmap.gesture_event[1]);
+		}
+	}
+#endif
+
+	data->input_serviced = false;
+	setup_int(data->dev, true);
+}
+
+int iqs5xx_init_interrupt(const struct device *dev)
+{
+	struct iqs5xx_data *data = dev->data;
+	const struct iqs5xx_dev_config *cfg = dev->config;
+	int rc;
+
+	if (!device_is_ready(cfg->rdy_gpio.port)) {
+		LOG_ERR("%s: device %s is not ready", dev->name, cfg->rdy_gpio.port->name);
+		return -ENODEV;
+	}
+
+	rc = gpio_pin_configure_dt(&cfg->rdy_gpio, GPIO_INPUT | cfg->rdy_gpio.dt_flags);
```

Comment:
nit: you've used variable names ret, rc, error in different places. Maybe just use ret everywhere?